bundles/specmate-migration/src/com/specmate/migration/internal/services/MigratorService.java

```diff
@@ -87,8 +87,10 @@ private void closeConnection() throws SpecmateException {
 
 	@Override
 	public boolean needsMigration() throws SpecmateException {
-		if (!initiateDBConnection()) {
-			logService.log(LogService.LOG_WARNING, "Migration: Could not obtain connection. Does the database exist?");
+		try {
+			initiateDBConnection();
+		} catch (SpecmateException e) {
+			logService.log(LogService.LOG_WARNING, e.getMessage());
 			return false;

```

Comment:
also here i would rather rethrow the exception to make clear that something bad happened. If we return false, the client will assume all is fine and no migration is needed, when in fact a migration might be needed but we just cannot determine it. Probably we should't even start Specmate if we cannot detmine if a migration is needed.

Reply:
I've simplified the logic and treat everything as an error (that should be handled by the client), except the case when the database does not exist. See source code in last commit for an explanation.